### PR TITLE
add DB2 provider documentation

### PR DESCRIPTION
**Pull request by dwadler**
We are reviewing the DB2 provider code internally and hope to have it ready to be integrated this week.

In the meantime, we added a section to Supported Data Formats of the User Manual to describe this functionality.

This pull request also includes two .png files for the new DB2 icons and updates to scripts/subst.py and source/conf.py to include the new icons.

### CODE REVIEW

**Comment by yjacolin** [comment]
Note to  Doc dev: Do not merge this PR until 2.14 is released as this feature is for 2.16. Thanks. :)

@dwadler thanks for your PR :) That's great to write code then documentation ;)

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 1]
```diff
@@ -825,3 +825,132 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+`DB2 LUW KnowledgeCenter`_.
+
+For more information about working with the DB2 spatial capabilities, check out
+the `DB2 Spatial Tutorial`_ on IBM DeveloperWorks.
+
+Creating a stored Connection
+............................
+
+|mActionAddDb2Layer| The DB2 provider uses ODBC to connect to a DB2 database.
+Windows includes ODBC by default. On *IX systems the unixodbc package needs to be
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: *IX means for rst that an emphasis started so the build failed. You should probably use `\*IX` or just use `UNIX like` term.

**Comment by dwadler** [comment]
Changed offending markup.

Where can I see the problem in the Travis testing?  I clicked on the "Details" link but didn't see anything useful.

**Comment by DelazJ** [comment]
At the end of the [report](https://travis-ci.org/qgis/QGIS-Documentation/builds/114576291)?

>Warning, treated as error:

>/home/travis/build/qgis/QGIS-Documentation/source/docs/user_manual/working_with_vector/supported_data.rst:860: WARNING: Inline emphasis start-string without end-string.

**Comment by dwadler** [comment]
Now that the DB2 provider has been integrated into Master, could the documentation update also be included?

**Comment by DelazJ** [comment]
The current official doc version of QGIS is 2.8 :(
We are currently working to publish a 2.14 version so that users may have a more recent documentation. Given that DB2 is not available in that version (but in next 2.16), your PR can't be merged in the 2.14 Doc. Once we published the 2.14 documentation, don't worry your PR is merged.
Thanks for having written a doc next to a feature.

**Comment by DelazJ** [comment]
fix #949

**Comment by yjacolin** [comment]
@dwadler could you rebase your PR from master? It is time to merge your contribution :)

**Comment by dwadler** [comment]
I merged master with DB2-Doc and pushed the changes to my fork but I'm not seeing the changes when I look at
https://github.com/dwadler/QGIS-Documentation/blob/DB2-Doc/source/conf.py

**Comment by dwadler** [comment]
It looks like my changes are in db2-doc, not DB2-Doc. Will close this and create a pull request from db2-doc.

**Pull request closed without merging**